hoomd/hpmc/UpdaterShape.h

```diff
@@ -0,0 +1,500 @@
+    GPUArray<unsigned int> m_ntypes;  // number of particle types in the simulation
+    bool m_initialized;               // whether or not the updater has been initialized
+    std::vector<Scalar> m_step_size;  // shape move stepsize
+    std::vector<unsigned int> m_count_accepted; // number of accepted updater moves
+    std::vector<unsigned int> m_count_total;    // number of attempted updater moves
+    std::vector<unsigned int>
+        m_box_accepted; // number of accepted moves between boxes in multi-phase simulations
+    std::vector<unsigned int>
+        m_box_total; // number of attempted moves between boxes in multi-phase simulations
+    detail::UpdateOrder m_update_order;      // order of particle types to apply the updater to
+    unsigned int m_instance = 0;             //!< Unique ID for RNG seeding
+    static constexpr Scalar m_tol = 0.00001; // minimum move size required to not be ignored.
+    };
+
+template<class Shape>
+UpdaterShape<Shape>::UpdaterShape(std::shared_ptr<SystemDefinition> sysdef,
+                                  std::shared_ptr<Trigger> trigger,
+                                  std::shared_ptr<IntegratorHPMCMono<Shape>> mc,
+                                  std::shared_ptr<ShapeMoveBase<Shape>> move)
+    : Updater(sysdef, trigger), m_mc(mc), m_move_function(move), m_global_partition(0),
+      m_determinant_inertia_tensor(m_pdata->getNTypes(), m_exec_conf),
+      m_ntypes(m_pdata->getNTypes(), m_exec_conf), m_initialized(false)
+    {
+    m_step_size.resize(m_pdata->getNTypes(), 0);
+    m_count_accepted.resize(m_pdata->getNTypes(), 0);
+    m_count_total.resize(m_pdata->getNTypes(), 0);
+    m_box_accepted.resize(m_pdata->getNTypes(), 0);
+    m_box_total.resize(m_pdata->getNTypes(), 0);
+    initializeDeterminatsInertiaTensor();
+    countParticlesPerType();
+
+    // TODO: add a sanity check to makesure that MPI is setup correctly
+    if (m_multi_phase)
+        {
+#ifdef ENABLE_MPI
+        MPI_Comm_rank(MPI_COMM_WORLD, &m_global_partition);
+        assert(m_global_partition < 2);
+#endif
+        }
+    }
+
+template<class Shape> UpdaterShape<Shape>::~UpdaterShape()
+    {
+    m_exec_conf->msg->notice(5) << "Destroying UpdaterShape " << std::endl;
+    }
+
+/*! Perform Metropolis Monte Carlo shape deformations
+\param timestep Current time step of the simulation
+*/
+template<class Shape> void UpdaterShape<Shape>::update(uint64_t timestep)
+    {
+    m_exec_conf->msg->notice(4) << "UpdaterShape update: " << timestep
+                                << ", initialized: " << std::boolalpha << m_initialized << " @ "
+                                << std::hex << this << std::dec << std::endl;
+    if (!m_initialized)
+        initializeDeterminatsInertiaTensor();
+
+    // Shuffle the order of particles types for this step
+    m_update_order.resize(m_pdata->getNTypes());
+    m_update_order.shuffle(timestep, this->m_sysdef->getSeed(), m_exec_conf->getRank());
```

Comment:
```suggestion
    m_update_order.shuffle(timestep, this->m_sysdef->getSeed(), RNGIdentifier::HPMCShapeMoveUpdateOrder);
```
Don't perform shape moves for different types on different ranks. Instead, introduce a new `RNGIdentifier` so that all ranks perform shape moves on the same type.